wazero/wazero.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
+)
+
+func main() {
+	doMain(os.Stdout, os.Stderr, os.Exit)
+}
+
+// doMain is separated out for the purpose of unit testing.
+func doMain(stdOut io.Writer, stdErr io.Writer, exit func(code int)) {
+	flag.CommandLine.SetOutput(stdErr)
+
+	var help bool
+	flag.BoolVar(&help, "h", false, "print usage")
+
+	flag.Parse()
+
+	if help {
+		printUsage(stdErr)
+		exit(0)
+	}
+
+	if flag.NArg() < 1 {
+		fmt.Fprintln(stdErr, "missing path to wasm file")
+		printUsage(stdErr)
+		exit(1)
+	}
+
+	var wasmArgs []string
+	if flag.NArg() > 1 {
+		if arg := flag.Arg(1); arg != "--" {
+			fmt.Fprintf(stdErr, "invalid argument: %s\n", arg)
+			printUsage(stdErr)
+			exit(1)
+		}
+		wasmArgs = flag.Args()[2:]
+	}
+
+	wasmPath := flag.Arg(0)
+	wasm, err := os.ReadFile(wasmPath)
+	if err != nil {
+		fmt.Fprintf(stdErr, "error reading wasm binary: %v\n", err)
+		printUsage(stdErr)
+		exit(1)
+	}
+
+	wasmExe := filepath.Base(wasmPath)
+
+	ctx := context.Background()
+	rt := wazero.NewRuntime(ctx)
+	defer rt.Close(ctx)
+
+	// Because we are running a binary directly rather than embedding in an application,
+	// we default to wiring up standard streams and args.
```

Comment:
do we want to plumb time and RNG? or wait till someone asks.

Reply:
Good call think it's good to plumb them now.